js/src/lib/interfaces/Libraries.ts

```diff
@@ -106,11 +106,41 @@ const asteroid = (model: ModelData) =>
 
 model = BaseModel.from_pretrained("${model.id}")`;
 
-const diffusers = (model: ModelData) =>
-	`from diffusers import DiffusionPipeline
+const diffusers_default = (model: ModelData) =>
+  `from diffusers import DiffusionPipeline
 
 pipeline = DiffusionPipeline.from_pretrained("${model.id}")`;
 
+const diffusers_controlnet = (model: ModelData) =>
+  `from diffusers import StableDiffusionControlNetPipeline, ControlNetModel
+
+controlnet = ControlNetModel.from_pretrained("${model.id}")
+pipeline = StableDiffusionControlNetPipeline.from_pretrained("${model.cardData?.base_model}", controlnet=controlnet)`;
```

Comment:
maybe handle the case where `model.cardData?.base_model` is undefined? or if it can't happen the types should reflect it

Reply:
(same for the other occurences)